bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,219 @@
+                                logger.debug("authentication to YIO dock ok");
+                                updateStatus(ThingStatus.ONLINE);
+                                try {
+                                    Runnable heartbeatpolling = new Runnable() {
+                                        @Override
+                                        public void run() {
+                                            try {
+                                                if (yioremotedockactualstatus
+                                                        .equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATED)) {
+                                                    yioremoteDockwebSocketClient
+                                                            .sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+
+                                                    Thread.sleep(1000);
+
+                                                    if (yioremoteDockwebSocketClient.getbooleanheartbeat()) {
+                                                        logger.debug("heartbeat ok");
+                                                        triggerChannel(getChannelUuid(GROUP_OUTPUT, YIODOCKSTATUS));
+                                                        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS,
+                                                                yioremoteDockwebSocketClient.getstringreceivedstatus());
+                                                    } else {
+                                                        yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTIONFAILED;
+                                                        updateStatus(ThingStatus.OFFLINE,
+                                                                ThingStatusDetail.CONFIGURATION_ERROR,
+                                                                "Connection lost no ping from YIO DOCK");
+                                                        updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                                                        pollingJob.cancel(true);
+                                                    }
+                                                } else {
+                                                    yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTIONFAILED;
+                                                    updateStatus(ThingStatus.OFFLINE,
+                                                            ThingStatusDetail.CONFIGURATION_ERROR,
+                                                            "Connection lost no ping from YIO DOCK");
+                                                    updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                                                    pollingJob.cancel(true);
+                                                }
+                                            } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.

Reply:
There is no better way for this unfortunately